File: at_coder/abc61-c.cc
#include <iostream>
#include <vector>
#include <map>
#include <iostream>

using namespace std;

class BigArray
{
private:
	long long n;
	long long k;
	map<long long, long long> mp;
public:
	BigArray(){
		cin >> n >> k;
		for (long long i = 0; i < n; ++i)
		{
			long long a;
			long long b;
			cin >> a >> b;
			mp[a] += b;
		}
	}
	void exec(){
		long long cur_i = 0;
		for (auto itr = mp.begin(); itr != mp.end(); ++itr)
		{
			cur_i += itr->second;
			if(cur_i >= k){
				cout << itr->first << endl; 
				return;
			}
		}
	}
};

int main(){
	BigArray ba = BigArray();
	ba.exec();
}

File: at_coder/abc61c.java
import java.util.Scanner;
import java.util.Arrays;
import java.util.TreeMap;
import java.util.Iterator;

public class abc61c {
	public static void main(String[] args) {
		BigArray bg = new BigArray();
		bg.exec();
	}
}

class BigArray {
	static long n;
	static long k;
	static TreeMap<Long, Long> map = new TreeMap<Long, Long>();

	public BigArray() {
		Scanner cin = new Scanner(System.in);
		this.n = cin.nextLong();
		this.k = cin.nextLong();
		for (int i = 0; i < n; i++) {
			long a = cin.nextLong();
			long b = cin.nextLong();
			if (map.containsKey(a)) {
				map.put(a, map.get(a) + b);
			} else {
				map.put(a, b);
			}
		}
	}

	void exec(){
		long cur_i = 0;
		for(long key : map.keySet()) {
			cur_i += map.get(key);
			if(cur_i >= k){
				System.out.println(key);
				return;
			}
		}
	}
}
